test/unit/plugins/providers/docker/action/prepare_networks_test.rb

```diff
@@ -343,7 +344,21 @@
       allow(driver).to receive(:network_containing_address).
         with("10.1.10.2").and_return("vagrant_network_public")
 
-      network_name, network_options = subject.process_public_network(options, {}, env)
+      # mock the call to PrepareNetworks.list_interfaces so that we don't depend
+      # on the current network interfaces
+      ifaddr = double("ifaddr")
+      addr = double("ip")
+      netmask = double("netmask")
+      allow(netmask).to receive(:ip_unpack).and_return(["255.255.255.0"])
+      allow(addr).to receive(:ip).and_return(true)
+      allow(addr).to receive(:ip_address).and_return("192.168.1.139")
+      allow(ifaddr).to receive(:addr).and_return(addr)
+      allow(ifaddr).to receive(:name).and_return("ens20u1u2")
+      allow(ifaddr).to receive(:netmask).and_return(netmask)
+      allow(subject).to receive(:list_interfaces).
+        and_return([ifaddr])
```

Comment:
So from the changes here, I _think_ we could actually simplify this and just simply have:

```ruby
allow(subject).to receive(:list_interfaces).
        and_return([ipaddr])
```

Notice the change from `ifaddr` to the double `ipaddr` defined above already.

I'm assuming maybe there are some missing fields though that we'd want to add to the double based on what you've `allow`'d above? You might also need to mock up `netmask` like you did....and if so I recommend defining it like we have above with the `let(:netmask)` way of defining doubles. Then instead of having allows everywhere you can do something like:

```ruby
let(:netmask) { double("netmask", ip_unpack: ["255.255.255.0"]) }
```

Similarly, you can probably add some missing fields to the double `ipaddr` above instead of the additional `allow`s.

Let me know if that works for you.

Reply:
Something like this?